### DISCUSSION TITLE
how can i set gas price and gas limit ?

### DISCUSSION THREAD

**Discussion by solidifilix** [lmusarella/Pancakeswap-Prediction-Bot-2023#34] [General]
please give me a code for add in project.i want set gas price to 3.00 gwei and gas limit to 330000

**Comment by jtruji68** [comment]
Hi, not the dev but it turns out the gas prices are set by your `BSC_NODE_PROVIDER_URL` in the `.env `file The ones i use are `"https://bscrpc.com"` and `"https://bsc-mainnet-rpc.allthatnode.com"` if you want to bet only in BSC. Those dont work very well with CAKE for some reason. Otherwise the data will be outdated and you cannot trade because the gas fees are not enough. If you find a provider that works well with CAKE please let us know. Also if you want to manually set the gas prices Im afraid you have to mess with the code directly, you can try with the info found in this reply :
[https://github.com/lmusarella/Pancakeswap-Prediction-Bot-2023/discussions/32#discussioncomment-7349888](url)
